src/rules/unbound-method.ts

```diff
@@ -73,9 +74,58 @@ export default createRule<Options, MessageIds>({
       return {};
     }
 
+    /**
+     * Checks if a MemberExpression is an argument to a `jest.mocked()` call.
+     * This handles cases like `jest.mocked(service.method)` where `service.method`
+     * should not be flagged as an unbound method.
+     */
+    const isArgumentToJestMocked = (
+      node: TSESTree.MemberExpression,
+    ): boolean => {
+      // Check if the immediate parent is a CallExpression
+      if (node.parent?.type !== AST_NODE_TYPES.CallExpression) {
+        return false;
+      }
+
+      const parentCall = node.parent;
+
+      // Check if the call is jest.mocked() by examining the callee
+      if (
+        parentCall.callee.type === AST_NODE_TYPES.MemberExpression &&
+        isSupportedAccessor(parentCall.callee.object) &&
```

Comment:
`isSupportedAccessor` takes an optional `value` as its second param for just these situations - so rather than having to do the work twice, you can just do `isSupportedAccessor(..., 'jest')`

that in turn means you should be able to refactor this all to be a `return`

Reply:
Updated.